### DISCUSSION TITLE
How to use automatic certificate management for SAML identity providers?

### DISCUSSION THREAD

**Discussion by jonasschwab** [keycloak/keycloak#27828] [Q&A]
If I understand [this changelog entry](https://www.keycloak.org/docs/24.0.0/release_notes/#automatic-certificate-management-for-saml-identity-providers) correctly, starting with version 24.0.0, I should be able to find a field "Metadata descriptor URL" in the settings of a SAML IdP and also a toggle "Use metadata descriptor URL". This would the regularly fetch and update IdP metadata.

Unfortuately, I cannot find these settings. Could someone point them out to me, or an error in my assumptions? Thank you!

**Comment by simonhanemann7882** [comment]
To configure Keycloak to use a metadata descriptor URL for a SAML Identity Provider (IdP), follow these steps:

1)  Log in to Keycloak Admin Console:
        Open your web browser and navigate to your Keycloak admin console URL.
        Log in with your admin credentials.

2)   Select the Realm:
        From the left-hand menu, select the realm where you want to configure the SAML IdP.

3)  Navigate to Identity Providers:
        In the left-hand menu, click on "Identity Providers".

4)  Add a SAML Identity Provider:
        Click on the "Add provider" dropdown and select "SAML v2.0".

5)   Configure the SAML IdP:
        You will be taken to the configuration page for the SAML Identity Provider.

6)  Set the Metadata Descriptor URL:
        Find the field labeled "Metadata URL" or "Metadata descriptor URL".
        Enter the URL where the IdP metadata is hosted.

 7) Enable Automatic Metadata Fetching:
        Look for an option labeled "Use metadata descriptor URL" or similar.
        Toggle this option to enable automatic fetching and updating of IdP metadata.

8)  Save the Configuration:
        Scroll down to the bottom of the page and click "Save".
